builtins/amp-img.js

```diff
@@ -82,9 +95,7 @@ export class AmpImg extends BaseElement {
     if (this.img_) {
       return;
     }
-    if (!this.srcset_) {
```

Comment:
if we are rewriting the parser, let's base it off of the blink implementation ( https://src.chromium.org/viewvc/blink/trunk/Source/core/html/parser/HTMLSrcsetParser.cpp#l264 ) lots of good test cases there as well.